### PR TITLE
Handle ValidateFormatString edge case

### LINKED ISSUE #42764: IDE0043 error is incorrect when using literal array
**Version Used**: 16.5.1

**Steps to Reproduce**:

```vb
Sub main()
    System.Console.WriteLine(String.Format("{0} {1}", {"hello", "world"}))
End Sub
```

**Expected Behavior**: No warnings

**Actual Behavior**: Warning	IDE0043	Format string contains invalid placeholder

### PR DESCRIPTION
**Pull request by sigmachirality**
fixes #42764

Small changes, adds some test cases to cover the different ways arrays can be defined in VB.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=45007) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=45007) before we can accept your contribution.<br/><br/>:x: sigmachirality [sign now](https://cla.dotnetfoundation.org/dotnet/roslyn?pullRequest=45007)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/roslyn?pullRequest=45007) it.</sub>

File: src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb [hunk 1]
```diff
@@ -27,11 +27,32 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.ValidateFormatStri
...
             Await TestDiagnosticMissingAsync("
 Class C
      Sub Main 
-        string.Format(""This {0} {1} {[||]2} works"", New Object  { ""test"", ""test2"", ""test3"" })
+        string.Format(""This {0} {1} {[||]2} works"", New Object() { ""test"", ""test2"", ""test3"" })
+    End Sub
+End Class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)>
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb:39]: could you add a WorkItem attribute to this?  thanks!

File: src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb [hunk 2]
```diff
@@ -27,11 +27,32 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.ValidateFormatStri
...
+        Public Async Function LiteralArray() As Task
+            Await TestDiagnosticMissingAsync("
+Class C
+     Sub Main 
+        string.Format(""This {0[||]} {1} {2} {3} works"", { ""test"", ""test2"", ""test3"", ""test4"" })
+    End Sub
+End Class")
+        End Function
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb:47]: could you make a C# equivalent test with `new [] { "test1", ... }`.  THanks!

File: src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb (hunk 1 above)
Comment (jmarolf) [reply src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb:39]: would look something like `<WorkItem(42764, "https://github.com/dotnet/roslyn/issues/42764")>`

File: src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb (hunk 2 above)
Comment (sigmachirality) [reply src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb:48]: added!

**Comment by CyrusNajmabadi** [comment]
Awesome, thanks :)

**Comment by CyrusNajmabadi** [comment]
Thanks much!

**Pull request merged**